src/blob/index/core.rs

```diff
@@ -244,9 +244,28 @@ where
         }
     }
 
-    fn mark_all_as_deleted_in_memory(headers: &mut InMemoryIndex<K>, key: &K) -> Option<u64> {
+    fn push_deletion_header_in_memory(
+        mem: &mut MemoryAttrs,
+        headers: &mut InMemoryIndex<K>,
+        key: &K,
+        header: RecordHeader,
+    ) -> Result<u64> {
+        mem.records_count += 1;
```

Comment:
I think it is better to do after the insertion happened (at the end of the function)